packages/js/data/src/crud/resolvers.ts

```diff
@@ -98,6 +105,16 @@ export const createResolvers = ( {
 	};
 
 	const getItemsTotalCount = function* ( query?: Partial< ItemQuery > ) {
+		const startedTotalCountUsingGetItems: boolean = yield controls.select(
+			storeName,
+			'hasStartedResolution',
+			`get${ pluralResourceName }`,
+			[ query ]
+		);
+		// Skip resolver as we get the total count from the getItems query as well with same query parameters.
+		if ( startedTotalCountUsingGetItems ) {
+			return;
+		}
```

Comment:
Skip the resolver request if we already have a `getItems` request started with the same query parameters. As this request already gives us the total count.